sorc/ncep_post.fd/UPP_PHYSICS.f

```diff
@@ -4497,6 +4500,483 @@ SUBROUTINE CALSLR_UUTAH(SLR)
+                  ABSV(I,J) = ((VWND(ip1,JJ)-VWND(im1,JJ))*wrk2(i,jj)         &
+     &                      +  (UWND(I,jj-1)*COSL(I,Jj-1)                     &
+     &                      -   UWND(I,J)*COSL(I,J))*wrk3(i,jj)) * wrk1(i,jj) &
+     &                      + F(I,Jj)
+                  DIV(I,J)  = ((UWND(ip1,JJ)-UWND(im1,JJ))*wrk2(i,jj)         &
+     &                      -  (VWND(I,jj-1)*COSL(I,Jj-1)                     &
+     &                      -   VWND(I,J)*COSL(I,J))*wrk3(i,jj)) * wrk1(i,jj) 
+                enddo
+              ENDIF
+            endif
+          ELSE
+            DO I=ISTA,IEND
+              ip1 = ie(i)
+              im1 = iw(i)
+              if(VWND(ip1,J)==SPVAL .or. VWND(im1,J)==SPVAL .or. &
+                 UWND(I,J-1)==SPVAL .or. UWND(I,J+1)==SPVAL) cycle
+              ABSV(I,J)   = ((VWND(ip1,J)-VWND(im1,J))*wrk2(i,j)               &
+     &                    -  (UWND(I,J-1)*COSL(I,J-1)                          &
+                          -   UWND(I,J+1)*COSL(I,J+1))*wrk3(i,j)) * wrk1(i,j)  &
+                          + F(I,J)
+              DIV(I,J)    = ((UWND(ip1,J)-UWND(im1,J))*wrk2(i,j)               &
+     &                    +  (VWND(I,J-1)*COSL(I,J-1)                          &
+                          -   VWND(I,J+1)*COSL(I,J+1))*wrk3(i,j)) * wrk1(i,j)  
+            ENDDO
+          END IF
+          if (npass > 0) then
+            do i=ista,iend
+              tx1(i) = absv(i,j)
+            enddo
+            do nn=1,npass
+              do i=ista,iend
+                tx2(i+1) = tx1(i)
+              enddo
+              tx2(1)    = tx2(im+1)
+              tx2(im+2) = tx2(2)
+              do i=2,im+1
+                tx1(i-1) = 0.25 * (tx2(i-1) + tx2(i+1)) + 0.5*tx2(i)
+              enddo
+            enddo
+            do i=ista,iend
+              absv(i,j) = tx1(i)
+            enddo
+          endif
+        END DO                               ! end of J loop
+
+!       deallocate (wrk1, wrk2, wrk3, cosl)
+! GFS use lon avg as one scaler value for pole point
+
+      ! call poleavg(IM,JM,JSTA,JEND,SMALL,COSL(1,jsta),SPVAL,ABSV(1,jsta))
+
+        call exch(absv(ista_2l:iend_2u,jsta_2l:jend_2u))
+        call fullpole(absv(ista_2l:iend_2u,jsta_2l:jend_2u),avpoles)     
+
+        cosltemp=spval
+        if(jsta== 1) cosltemp(1:im, 1)=coslpoles(1:im,1)
+        if(jend==jm) cosltemp(1:im,jm)=coslpoles(1:im,2)
+        avtemp=spval
+        if(jsta== 1) avtemp(1:im, 1)=avpoles(1:im,1)
+        if(jend==jm) avtemp(1:im,jm)=avpoles(1:im,2)
+        
+        call poleavg(IM,JM,JSTA,JEND,SMALL,cosltemp(1,jsta),SPVAL,avtemp(1,jsta))
+
+        if(jsta== 1) absv(ista:iend, 1)=avtemp(ista:iend, 1)
+        if(jend==jm) absv(ista:iend,jm)=avtemp(ista:iend,jm)
+    
+!        deallocate (wrk1, wrk11, wrk2, wrk3, cosl, iw, ie)
+
+    ENDIF  ! END of MODELNAME=='GFS' BLOCK
+    
+      call exch(absv(ista_2l:iend_2u,jsta_2l:jend_2u))
```

Comment:
If model is not 'GFS' or global ABSV and DIV will be SPVAL at this moment. Are computations after this line correct if ABSV and DIV are SPVAL?

Reply:
@KarinaAsmar-NOAA @JesseMeng-NOAA After the modelname condition for 'GFS' is applied in MDL2P.f, this IF state might not be needed.